src/natcap/invest/ndr/ndr.py

```diff
@@ -949,7 +949,9 @@ def _map_load_op(lucode_array):
         result = numpy.empty(lucode_array.shape)
         result[:] = _TARGET_NODATA
         for lucode in numpy.unique(lucode_array):
-            if lucode != nodata_landuse:
+            lucode_is_nodata = utils.array_equals_nodata(
+                numpy.array([lucode]), nodata_landuse).any()
+            if not lucode_is_nodata:
```

Comment:
Yes, totally. I ran into this problem of, how far back in the code should I check so that I can be certain of the array type, vs, one of the functions purposes is to help take the ambiguity out of it. But, it did make me wonder if we should just be more careful about how we handle the model input rasters in operations (with things like the code snippet you have) and in defining constant OUT_FLOAT_NODATA values, so that in downstream operations, we would know we didn't have to worry about handling `nan` cases.